### DISCUSSION TITLE
Minimal pom.xml content for building OpenCV app based on snapshots?

### DISCUSSION THREAD

**Discussion by JeeDevUser** [bytedeco/javacv#1814] [Q&A] [answered]
Hi,
I am just developing OpenCV  Java application, which uses the 
https://docs.opencv.org/4.x/df/d38/classcv_1_1ccm_1_1ColorCorrectionModel.html

And JavaCV currently support this:
https://github.com/bytedeco/javacpp-presets/commit/f575bf0739f425b7e8d210555a77e9b2b3b17f86
however, this is included in the snapshots build 

Now, here is my pom.xml content:
http://bytedeco.org/builds/

Wondering, what is a minimal pom.xml content, for building OpenCV application, based on snapshots?
it seems to me that, with my current pom.xml content,  I take a lot of things that I don't need at all...
I would like to take only what I need for OpenCV app...

**Comment by saudet** [comment, marked as answer]
This sample pom.xml file here depends only on OpenCV:
https://github.com/bytedeco/javacpp-presets/blob/master/opencv/samples/pom.xml
Like I keep telling you, stuff related to the presets are all on that repository.

**Comment by JeeDevUser** [comment]
Thanks!

**Reply by JeeDevUser** [reply]
@saudet  , when I used mentioned pom.xml, I am getting this errors:

_Missing artifact org.bytedeco:opencv-platform-gpu:jar:4.5.5-1.5.8-SNAPSHOT
Missing artifact org.bytedeco:cuda-platform-redist:jar:11.6-8.3-1.5.8-SNAPSHOT
Missing artifact org.bytedeco:mkl-platform-redist:jar:2022.0-1.5.8-SNAPSHOT
Missing artifact org.bytedeco:numpy-platform:jar:1.22.3-1.5.8-SNAPSHOT_

And when I remove all optional dependencies, leaving just

        <dependency>
            <groupId>org.bytedeco</groupId>
            <artifactId>opencv-platform</artifactId>
            <version>4.5.5-1.5.8-SNAPSHOT</version>
        </dependency>

        
I am getting this:

**_The import org.bytedeco.javacv cannot be resolved_**

Can you help?
Thanks in advance!

**Reply by JeeDevUser** [reply]
this one seems to have solved the problem (forgot to mention I am using javacv also :-(

	_<dependency>
		<groupId>org.bytedeco</groupId>
		<artifactId>javacv</artifactId>
		<version>1.5.7</version>
	</dependency>_
	

whether this is the minimum needed to solve following import:
_**import org.bytedeco.javacv.OpenCVFrameConverter;**_

...huh?

**Comment by saudet** [comment]
Yes, that one works for that.

**Reply by JeeDevUser** [reply]
@saudet 
Yes. 
i just made a jar file, which includes only above two dependencies, by using _maven assembly plugin_, as described here:
https://stackoverflow.com/questions/60843834/with-maven-how-to-bundle-all-dependencies-and-my-java-project-filesin-the-output

But the file size is about **513 MB!**

Here is all Java imports used by app (plain Java desktop app),, you see, there are only a few :

`import org.bytedeco.opencv.opencv_core.*;
import org.bytedeco.opencv.opencv_imgproc.*;
import org.bytedeco.opencv.opencv_mcc.CChecker;
import org.bytedeco.opencv.opencv_mcc.CCheckerDetector;
import org.bytedeco.opencv.opencv_mcc.CCheckerDraw;
import org.bytedeco.opencv.opencv_mcc.ColorCorrectionModel;
import org.opencv.core.Core;
import org.opencv.core.CvType;
import org.opencv.highgui.HighGui;
import org.opencv.imgcodecs.Imgcodecs;
import org.opencv.imgproc.Imgproc;
import static org.bytedeco.opencv.global.opencv_core.*;
import static org.bytedeco.opencv.global.opencv_imgproc.*;
import org.bytedeco.javacv.OpenCVFrameConverter;
import org.opencv.calib3d.Calib3d;
import org.opencv.core.Core;
import org.opencv.core.CvType;
import org.opencv.core.Mat;
import org.opencv.core.MatOfByte;
import org.opencv.core.MatOfDMatch;
import org.opencv.core.MatOfKeyPoint;
import org.opencv.core.MatOfPoint;
import org.opencv.core.MatOfPoint2f;
import org.opencv.core.Point;
import org.opencv.features2d.DescriptorMatcher;
import org.opencv.features2d.Features2d;
import org.opencv.features2d.SIFT;
import org.opencv.highgui.HighGui;
import org.opencv.imgcodecs.Imgcodecs;
import org.opencv.imgproc.Imgproc;

import org.opencv.core.DMatch;
import org.opencv.core.KeyPoint;
import org.opencv.core.Scalar;
import org.opencv.core.Size;


_**I wonder, is there a way to make a smaller ,jar file ?**_
The app will not run on Andoid, nor Linux, I just need support WIndows platform

**Reply by saudet** [reply]
Yes: https://github.com/bytedeco/javacpp-presets/wiki/Reducing-the-Number-of-Dependencies

**Reply by JeeDevUser** [reply]
@saudet , you saved my day !
Thanks a lot!  and sorry again because I still don't know where to ask the question.
All this is still new to me, javacv, presets, opencv,....a lot of things, really...
